### ISSUE TITLE
Regression (rustc): Formatting UFCS function calls produces invalid code (needstest)

### ISSUE THREAD

**Issue by pyfisch** [nrc/rustfmt#64]
```rust
fn header_name<T: Header>() -> &'static str {
    let name = <T as Header>::header_name();
    name
}
```
is formatted to
```rust
fn header_name<T: Header>() -> &'static str {
    let name = <();
    name
}
```

which is clearly not intended.

**Comment by defyrlt** [comment]
I ran this through idempotent tests and got this:
```rust
fn header_name<T: Header>() -> &'static str {
    let name = <T as Header>::header_name(();
    name
}
```

**Comment by nrc** [comment]
Given that this is caused by a regression in rustc (by the sounds of it), I'm going to put money on this being due to an incorrect span from the parser.

**Comment by marcusklaas** [comment]
I've opened an issue for this here: https://github.com/rust-lang/rust/issues/25969.

**Comment by marcusklaas** [comment]
The underlying issue has been resolved: https://github.com/rust-lang/rust/pull/25994. I can confirm that the test case above is now formatted correctly when using the master branch of rustc.

**Comment by nrc** [comment]
This just needs a test now. (thanks for fixing the rustc bug @marcusklaas)

**Comment by nrc** [comment]
Closed by #97
